p2p/profile_service.go

```diff
@@ -0,0 +1,227 @@
+package p2p
```

Comment:
Generally, we should aim to get a local test for this code up as well as the testground work. We can do that in a follow-up